Update templates-ref.rst to mention `extras` and format Airflow Vars / Conns (#24735)
mistercrunch/Airflow: docs/apache-airflow/templates-ref.rst

Refer to the models documentation for more information on the objects'
attributes and methods.

> Airflow Variables in Templates
> ---------

Reviewer: ```suggestion
Airflow Variables in Templates
------------------------------
```

> The ``var`` template variable allows you to access variables defined in Airflow's
> UI. You can access them as either plain-text or JSON. If you use JSON, you are

Reviewer 2: ```suggestion
The ``var`` template variable allows you to access Airflow Variables.
You can access them as either plain-text or JSON. If you use JSON, you are
```

Might as well fix this while we are in here.

``{{ var.json.get('my.dict.var', {'key1': 'val1'}) }}``. Defaults can be
supplied in case the variable does not exist.

Airflow Connections in Templates

> ---------

Reviewer: ```suggestion
Airflow Connections in Templates
---------------------------------
```

The line has to match the heading in terms of length, it can be more not less I think

Similarly, Airflow Connections data can be accessed via the ``conn`` template variable. For example, you could use expressions in your templates like ``{{ conn.my_conn_id.login }}``,
``{{ conn.my_conn_id.password }}``, etc.

Just like with ``var`` it's possible to fetch a connection by string  (e.g. ``{{ conn.get('my_conn_id_'+index).host }}``
) or provide defaults (e.g ``{{ conn.get('my_conn_id', {"host": "host1", "login": "user1"}).host }}``).

> Additionally, the `extras` field of a connection can be fetched as a Python Dictionary by the ``extra_dejson`` field, e.g. 

Reviewer 2: ```suggestion
Additionally, the ``extras`` field of a connection can be fetched as a Python Dictionary with the ``extra_dejson`` field, e.g. 
```

rst need doubles, and "with" sounds better than "by" to my ear.

> Additionally, the ``extras`` field of a connection can be fetched as a Python Dictionary with the ``extra_dejson`` field, e.g. 

Reviewer 2: ```suggestion
Additionally, the ``extras`` field of a connection can be fetched as a Python Dictionary with the ``extra_dejson`` field, e.g.
```
